client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDatabaseMetaData.java

```diff
@@ -1485,6 +1519,17 @@ private static void optionalStringInFilter(List<String> filters, String columnNa
         filters.add(filter.toString());
     }
 
+    @Nullable
+    private String escapeIfNecessary(@Nullable String namePattern)
+    {
+        if (namePattern == null || !connection.isAssumeLiteralNamesInMetadataCalls()) {
+            return namePattern;
+        }
+        //noinspection ConstantConditions
+        verify(SEARCH_STRING_ESCAPE.equals("\\"));
+        return namePattern.replaceAll("[_%\\\\]", "\\$1");
```

Comment:
Probably deserves a comment describing what the intent is.

Reply:
The regex replace needs some unit testing too, be wary of this from the docs: 

"Note that backslashes (\) and dollar signs ($) in the replacement string may cause the results to be different than if it were being treated as a literal replacement string. Dollar signs may be treated as references to captured subsequences as described above, and backslashes are used to escape literal characters in the replacement string"